tasks/module_4/task_2.py

```diff
@@ -0,0 +1,51 @@
+"""Module4/task2: File Extension Handler."""
+import os
+from contextlib import contextmanager
+
+
+FILE_PREFIX = 'spam'
+FILES_DIR = 'resources/task_2'
+
+
+@contextmanager
+def cwd(new_dir):
+    """Custom-made context manager for temporarily directory switching during runtime."""
+    old_dir = os.getcwd()
+    os.chdir(new_dir)
+    try:
+        yield
+    finally:
+        os.chdir(old_dir)
+
+
+def get_files_with_prefix() -> list:
+    """Get files with a specified prefix.
+
+    :return list of files matching a given prefix.
+    """
+    matching_files = []
+    for file_ in os.listdir():
```

Comment:
Nice job with `_` 👍

Reply:
And you may also take a look at [os.scandir](https://docs.python.org/3/library/os.html#os.scandir). It is more performant when you have a lot of entries in the directory. It is not your case though.